rewrite/rewrite/python/printer.py

```diff
@@ -0,0 +1,1350 @@
+        p.append("in")
+        self.visit(clause.iterated_list, p)
+        if clause.conditions is not None:
+            for condition in clause.conditions:
+                self.visit(condition, p)
+        return clause
+
+    def visit_type_alias(self, type_alias: TypeAlias, p: PrintOutputCapture[P]) -> J:
+        self.before_syntax(type_alias, PySpace.Location.UNION_TYPE_PREFIX, p)
+        p.append("type")
+        self.visit(type_alias.name, p)
+        self._visit_left_padded_printer("=", type_alias.padding.value, PyLeftPadded.Location.TYPE_ALIAS_VALUE, p)
+        self.after_syntax(type_alias, p)
+        return type_alias
+
+    def visit_yield_from(self, yield_from: YieldFrom, p: PrintOutputCapture[P]) -> J:
+        self.before_syntax(yield_from, PySpace.Location.YIELD_FROM_PREFIX, p)
+        p.append("from")
+        self.visit(yield_from.expression, p)
+        return yield_from
+
+    def visit_union_type(self, union_type: UnionType, p: PrintOutputCapture[P]) -> J:
+        self.before_syntax(union_type, PySpace.Location.UNION_TYPE_PREFIX, p)
+        # TODO: Check name inconsistency between java and python UNION_TYPE_TYPES vs UNION_TYPE_TYPE
+        self._visit_right_padded_printer(union_type.padding.types, PyRightPadded.Location.UNION_TYPE_TYPES, "|", p)
+        self.after_syntax(union_type, p)
+        return union_type
+
+    def visit_variable_scope(self, variable_scope: VariableScope, p: PrintOutputCapture[P]) -> J:
+        self.before_syntax(variable_scope, PySpace.Location.VARIABLE_SCOPE_PREFIX, p)
+        if VariableScope.Kind.GLOBAL == variable_scope.kind:
+            p.append("global")
+        elif VariableScope.Kind.NONLOCAL == variable_scope.kind:
+            p.append("nonlocal")
+
+        # TODO: Check name inconsistency between java and python VARIABLE_SCOPE_ELEMENTS vs VARIABLE_SCOPE_NAME_SUFFIX
+        self._visit_right_padded_printer(variable_scope.padding.names, PyRightPadded.Location.VARIABLE_SCOPE_NAMES, ",",
+                                         p)
+        return variable_scope
+
+    def visit_del(self, del_: Del, p: PrintOutputCapture[P]) -> J:
+        self.before_syntax(del_, PySpace.Location.DEL_PREFIX, p)
+        p.append("del")
+        # Note: PythonPrinter.java uses PyRightPadded.Location.DEL_ELEMENTS here but that doesn't exist in Python.
+        # Instead we use PyRightPadded.Location.DEL_TARGETS as both actually point to PySpace.Location.DEL_TARGET_SUFFIX.
+        self._visit_right_padded_printer(del_.padding.targets, PyRightPadded.Location.DEL_TARGETS, ",", p)
+        return del_
+
+    def visit_special_parameter(self, special_parameter: SpecialParameter, p: PrintOutputCapture[P]) -> J:
+        self.before_syntax(special_parameter, PySpace.Location.SPECIAL_PARAMETER_PREFIX, p)
+        if special_parameter.kind == 'ARGS':
+            p.append("*")
+        elif special_parameter.kind == 'KWARGS':
+            p.append("**")
```

Comment:
I don't think this is correct. We have defined an enum `SpecialParameter.Kind` and should compare against its constants.